app/controllers/email/BouncedEmailController.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.email
+
+import com.google.inject.Inject
+import common.SessionKeys
+import config.{AppConfig, ErrorHandler}
+import controllers.BaseController
+import controllers.predicates.AuthPredicateComponents
+import controllers.predicates.inflight.InFlightPredicateComponents
+import forms.YesNoForm.yesNoForm
+import models.{No, User, Yes}
+import play.api.mvc.{Action, AnyContent}
+import services.VatSubscriptionService
+import utils.LoggerUtil
+
+import scala.concurrent.ExecutionContext
+
+class BouncedEmailController @Inject()(val errorHandler: ErrorHandler,
+                                       val subscriptionService: VatSubscriptionService)
+                                      (implicit val authComps: AuthPredicateComponents,
+                                       inFlightComps: InFlightPredicateComponents,
+                                       appConfig: AppConfig) extends BaseController with LoggerUtil {
+
+  implicit val ec: ExecutionContext = authComps.mcc.executionContext
+
+  def show: Action[AnyContent] = blockAgentPredicate.async { implicit user =>
+    subscriptionService.getCustomerInfo(user.vrn) map {
+      case Right(details) => details.ppob.contactDetails.flatMap(_.emailVerified) match {
+        case Some(true) => Redirect(appConfig.vatOverviewUrl)
+        case _ =>
+          extractSessionEmail match {
+            case Some(email) => Ok(email)
+            case _ => Redirect(routes.CaptureEmailController.show)
+          }
+      }
+      case _ => errorHandler.showInternalServerError
+    }
+  }
+
+  def submit: Action[AnyContent] = blockAgentPredicate { implicit request =>
+    yesNoForm("error").bindFromRequest.fold(
+      _ => errorHandler.showInternalServerError,
```

Comment:
I'm a bit worried a line like this may be overlooked in the wiring up. The switch of form is the most obvious change here but I wonder would the person doing the work know they need to change this error to the appropriate 400 response & page reload? I think either some TODOs in the code or even just notes on the wiring up Jira subtask would help to make it clear what is temporary here.